packages/cli/api-importers/openapi/openapi-ir-parser/src/schema/convertAdditionalProperties.ts

```diff
@@ -104,20 +105,68 @@ export function convertAdditionalProperties({
             }),
             groupName: undefined
         },
-        valueSchema: convertSchema(
-            additionalProperties,
-            context.options.optionalAdditionalProperties ? wrapAsNullable : false,
-            context,
-            [...breadcrumbs, "Value"],
-            source,
-            namespace
+        // Whether a type is inline is usually determined later by checking if a declaration is nested within another declaration,
+        // however this map is generated from the additionalProperties and thus is inline of the object (depending on the value type),
+        // so we allways add inline (depending on the value type).
+        valueSchema: addInline(
+            convertSchema(
+                additionalProperties,
+                context.options.optionalAdditionalProperties ? wrapAsNullable : false,
+                context,
+                [...breadcrumbs, "Value"],
+                source,
+                namespace,
+                undefined,
+                undefined,
+                undefined
+            )
         ),
         groupName,
         example,
         encoding
     });
 }
 
+function addInline(schema: SchemaWithExample): SchemaWithExample {
+    switch (schema.type) {
+        case "array":
+            schema.inline = true;
+            break;
+        case "enum":
+            schema.inline = true;
+            break;
+        case "literal":
+            break;
+        case "map":
+            schema.inline = true;
+            break;
+        case "nullable":
+            schema.inline = true;
+            schema.value = addInline(schema.value);
+            break;
+        case "object":
+            schema.inline = true;
+            break;
+        case "oneOf":
+            schema.value.inline = true;
+            break;
+        case "optional":
+            schema.inline = true;
+            schema.value = addInline(schema.value);
+            break;
+        case "primitive":
+            break;
+        case "reference":
+            break;
+        case "unknown":
+            break;
+        default:
+            assertNever(schema);
+    }
+
+    return schema;
+}
```

Comment:
nit: You can group these cases together for slightly better readability. We also don't need the final `return` if we use `assertNever` like you already are.

```suggestion
function addInline(schema: SchemaWithExample): SchemaWithExample {
    switch (schema.type) {
        case "array":
        case "enum":
        case "map":
        case "object":
        case "oneof":
            schema.inline = true;
            break;
        case "nullable":
        case "optional":
            schema.inline = true;
            schema.value = addInline(schema.value);
            break;
        case "literal":
        case "primitive":
        case "reference":
        case "unknown":
            break;
        default:
            assertNever(schema);
    }
}
```

Reply:
I think eslint didn't let me do this, but I'll try again